libp2p/NodeTable.h

```diff
@@ -306,7 +320,13 @@ class NodeTable : UDPSocketEvents
 
     bool m_allowLocalDiscovery;                                     ///< Allow nodes with local addresses to be included in the discovery process
 
-    DeadlineOps m_timers; ///< this should be the last member - it must be destroyed first
+    // timers are shared_ptr so we can keep them alive if handlers are executed after the node table
+    // has been destroyed so we can check for timer cancellation. Note that this is technically not
```

Comment:
Well yeah I think I got rid of this `enable_shared_from_this` approach a while ago, because we clearly control the life time of the `NodeTable` from the level above (`Host`) and should be able to cancel the timers when it's destructed.
This comeback doesn't seem to make the code simpler...